### PR TITLE
Request lists and edits

### PR DESCRIPTION
**Pull request by nyet-ty**
## PR includes

-   [x] Bug Fix
-   [x] Refactor
-   [x] Feature

### CODE REVIEW

File: src/modules/userCreationRequestMethods.ts [hunk 1]
```diff
@@ -611,10 +701,36 @@ export const userCreationRequestsMethods = {
     getList: async (data: GetUserCreationRequestList): Promise<CompleteUserCreationRequest[]> => {
         const where: Prisma.UserCreationRequestWhereInput = {};
 
+        if (data.type) {
+            where.type = { in: data.type };
+        }
+
+        if (data.status !== undefined) {
```
Comment (tsumo) [inline src/modules/userCreationRequestMethods.ts:708]: `status !== undefined` ->  `where.status = null` looks suspicious

File: src/modules/userCreationRequestMethods.ts [hunk 2]
```diff
@@ -787,11 +902,13 @@ export const userCreationRequestsMethods = {
             workSpace: workSpace || undefined,
             personalEmail: personalEmail || undefined,
             workEmail: workEmail || undefined,
-            supplementalPositions: supplementalPositions.map(({ organizationUnitId, unitId, percentage }) => ({
-                organizationUnitId,
-                unitId: unitId || undefined,
-                percentage,
-            })),
+            supplementalPositions: supplementalPositions
+                .filter((position) => !position.main)
```
Comment (tsumo) [inline src/modules/userCreationRequestMethods.ts:906]: main position is not returned?

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/modules/userCreationRequestMethods.ts (hunk 2 above)
Comment (nyet-ty) [reply src/modules/userCreationRequestMethods.ts:906]: this field is only for additional position, main goes in organizationUnitId

File: src/modules/userCreationRequestMethods.ts (hunk 1 above)
Comment (nyet-ty) [reply src/modules/userCreationRequestMethods.ts:708]: we have two statuses: Accepted and Declined, if request on consideration, its status === null. By default we need to show only requests with nullish status

**Review by asabotovich — APPROVED** [review]

**Pull request merged**